app/scripts/controllers/overview.js

```diff
@@ -1434,11 +1436,17 @@ function OverviewController($scope,
     }, {poll: true, pollInterval: DEFAULT_POLL_INTERVAL}));
 
     if ($scope.AEROGEAR_MOBILE_ENABLED) {
-      watches.push(DataService.watch({ group: "mobile.k8s.io", version: "v1alpha1", resource: "mobileclients" }, context, function (clients) {
+      watches.push(MobileClientsService.watch(context, function(clients) {
         overview.mobileClients = clients.by("metadata.name");
+        _.each(overview.mobileClients , function (mobileClient) {
+          mobileClient.resourceData = {};
+          mobileClient.resourceData.icon = _.get(mobileClient, 'metadata.annotations.icon');
```

Comment:
use `_.set(mobileClient, 'resourceData.icon', _.get(mobileClient, 'metadata.annotations.icon'));` But not sure why you need to create new `resourceData` attribute of mobileClient object?

Reply:
This standardises the data (mobile service instance and mobile client) so that the mobile resource looks the same

Reply:
use `copiedClient=angular.copy(mobileClient);` when you want to modify something and not modify original `mobileClient` then pass `copiedClient` further